extensions/security-webauthn/runtime/src/main/java/io/quarkus/security/webauthn/WebAuthnRecorder.java

```diff
@@ -0,0 +1,77 @@
+package io.quarkus.security.webauthn;
+
+import java.security.SecureRandom;
+import java.util.Base64;
+import java.util.function.Supplier;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.runtime.BeanContainer;
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.runtime.annotations.Recorder;
+import io.quarkus.security.identity.IdentityProviderManager;
+import io.quarkus.vertx.http.runtime.HttpConfiguration;
+import io.quarkus.vertx.http.runtime.security.PersistentLoginManager;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.handler.BodyHandler;
+
+@Recorder
+public class WebAuthnRecorder {
+
+    private static final Logger log = Logger.getLogger(WebAuthnRecorder.class);
+
+    final RuntimeValue<HttpConfiguration> httpConfiguration;
+    final RuntimeValue<WebAuthnRunTimeConfig> config;
+
+    //the temp encryption key, persistent across dev mode restarts
+    static volatile String encryptionKey;
+
+    public WebAuthnRecorder(RuntimeValue<HttpConfiguration> httpConfiguration, RuntimeValue<WebAuthnRunTimeConfig> config) {
+        this.httpConfiguration = httpConfiguration;
+        this.config = config;
+    }
+
+    public void setupRoutes(BeanContainer beanContainer, RuntimeValue<Router> routerValue) {
+        WebAuthnSecurity security = beanContainer.instance(WebAuthnSecurity.class);
+        WebAuthnAuthenticationMechanism authMech = beanContainer.instance(WebAuthnAuthenticationMechanism.class);
+        IdentityProviderManager identityProviderManager = beanContainer.instance(IdentityProviderManager.class);
+        WebAuthnController controller = new WebAuthnController(security, config.getValue(), identityProviderManager, authMech);
+        Router router = routerValue.getValue();
+        BodyHandler bodyHandler = BodyHandler.create();
+        // FIXME: paths configurable
+        router.post("/webauthn/login").handler(bodyHandler).handler(controller::login);
+        router.post("/webauthn/register").handler(bodyHandler).handler(controller::register);
+        router.post("/webauthn/callback").handler(bodyHandler).handler(controller::callback);
+        router.get("/webauthn/webauthn.js").handler(controller::javascript);
+        router.get("/webauthn/logout").handler(controller::logout);
+    }
+
+    public Supplier<WebAuthnAuthenticationMechanism> setupWebAuthnAuthenticationMechanism() {
+
+        return new Supplier<WebAuthnAuthenticationMechanism>() {
+            @Override
+            public WebAuthnAuthenticationMechanism get() {
+                String key;
+                if (!httpConfiguration.getValue().encryptionKey.isPresent()) {
```

Comment:
I'm not sure it is a good idea to have that reused, I mean, it is unlikely people would prefer to use `FormAuth` (which is what that key configuration was intended for) alongside `WebAuthn`  - but nonetheless I think it will be cleaner to have a similar property in `WebAuthnRunTimeConfig`

Reply:
@FroMage but `WebAuthn` and `Form` are two different mechanisms which can co-exist, so even though both need the encryption key, it is generally not a good idea to use the same key material for different targets/purposes. The only thing, as I said, it appears very unlikely these 2 mechanisms can co-exist in a production but Quarkus allows for it. I'd really like @stuartwdouglas to comment on re-using the config/code intended for Form originally for WebAuthn. But it is a `preview` so I guess it is not a blocker for a merge

Reply:
From my PoV, every encrypted/signed cookie from Quarkus should use the same algo/key/settings. Because I really can't imagine why we'd want different settings for different cookies.